### PR TITLE
Guide to template files

### PR DESCRIPTION
**Pull request by sjcorbett**
A full guide to using template files in blueprints.

### CODE REVIEW

**Comment by sjcorbett** [comment]
@rdowner I think this is something you were looking at previously.

File: guide/blueprints/config-files.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+title: Uploading Script and Configuration Files
+layout: website-normal
+toc: ../guide_toc.json
+categories: [use, guide, defining-applications]
+---
+
+Blueprints often require that parameterized scripts and configuration files are available to be copied to the
+target VM. These must be URLs resolvable from the Brooklyn instance, or on the Brooklyn classpath. One simple way 
+to achieve this is to compile the support files into a .jar, which is then added to Brooklyn's 'dropins' folder. 
+Alternatively, an OSGi bundle can be used, referenced from the catalog item.
```
Comment (drigodwin) [inline guide/blueprints/config-files.md]: Perhaps mention that these methods are specific to the way brooklyn is launched.

File: guide/blueprints/config-files.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+template. This supports a powerful set of substitutions. In brief, anything (unescaped) of the form
+`${name}` will be substituted, in this case looking up "name" for the value to use.
+
+
+### Using files
+
+Files can be referenced as URLs. This includes support for things like `classpath://mypath/myfile.bat`. 
+This looks for the given (fully qualified) resource on the Brooklyn classpath.
+
+The destination for the file upload is specified in the entity's configuration. Note that the backslash character
```
Comment (drigodwin) [inline guide/blueprints/config-files.md]: What's the configuration key? or perhaps an example of one.

File: guide/blueprints/config-files.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+The value can be referenced in a template with `${databaseAddress}`.
+
+
+### Example
+
+    files.preinstall:
+      # Reference a fixed resource
+      classpath://com/acme/installAcme.ps1: C:\\acme\installAcme.ps1
+      # Inject the source from a config key
+      $brooklyn:config("acme.conf"): C:\\acme\acme.conf
```
Comment (rdowner) [inline guide/blueprints/config-files.md]: I'd also like to see described how to install files to INSTALL_DIR and/or RUN_DIR - do I just omit the path component?

File: guide/blueprints/config-files.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+| Substitution              | Effect                                                             |
+|---------------------------|--------------------------------------------------------------------|
+| `${config['key']}`        | Equivalent to `entity.config().get(key)`                           |
+| `${attribute['key']}`     | Equivalent to `entity.sensors().get(key)`                          |
+| `${mgmt['key']}`          | Loads the value for `key` from the management context's properties |
+| `${entity.foo}`           | FreeMarker calls `getFoo` on the entity                            |
+| `${driver.foo}`           | FreeMarker calls `getFoo` on the entity's [driver](http://brooklyn.apache.org/v/latest/java/entity.html#things-to-know) |
+| `${location.foo}`         | FreeMarker calls `getFoo` on the entity's location                 |
+| `${javaSysProps.foo.bar}` | Loads the system property named `foo.bar`                          |
```
Comment (rdowner) [inline guide/blueprints/config-files.md]: How do I insert a literal `${ABC}`? (Important in shell scripts, for example)

**Comment by sjcorbett** [comment]
@drigodwin @rdowner thanks. I've addressed all your comments.

File: guide/blueprints/config-files.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+To include the literal `${value}` in a script you might:
+ * specify a [raw string literal](http://freemarker.org/docs/dgui_template_exp.html#dgui_template_exp_direct_string):
+   `${r"${value}"}`
+ * use the [noparse](http://freemarker.org/docs/ref_directive_noparse.html) directive: `<#noparse>${value}</#noparse>`
+ * use FreeMarker's [alternative syntax](http://freemarker.org/docs/dgui_misc_alternativesyntax.html).
+
+A common pattern for templating Bash files is to set environment variables at the top of the script and to surround
+the rest of its contents with `noparse`. For example:
+
+```
```
Comment (drigodwin) [inline guide/blueprints/config-files.md]: The website uses kramdown not markdown, the three backticks doesn't display in the same way. It's better to use four spaces before each line as in the example section.

**Pull request merged**